src/lib/SelectOptionsTypes.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * A library to build select options for dropdowns
+ */
+
+/*
+ * @param {string} key - the key to use for the value of the option
+ * @param {string} name - the key to use for the text of the option
+ * @param {array} list - the list of items to filter and create options from
+ * @returns {object} - an object with fields, filter, and options
+ */
+const SelectOptionsType = ({ key, name, list }) => {
+  // fields is an object with the key and name to use for the value and text of the option
+  const fields = {
+    value: key,
+    text: name,
+  }
+
+  /*
+   * filter returns a list of items that are active or have the same key as the current item
+   * @param {object} currentItem - the current item to compare to
+   * @returns {array} - a list of items that are active or have the same key as the current item
+   */
+  const filter = (currentItem) => {
```

Comment:
I would change the name of this filter to something else as 'filter' is a JS keyword and it isn't descriptive of what it is doing. Alternatively you could make this function more agnostic to take the filters like 'active' and 'key' as parameters and call the function filterMatch or something.

Reply:
I did think about making it more flexible but would wait until we have further selects that could do with the same treatment. filter is not a JS keyword as far as I know. It is a method on Array which is why it would not be a problem here as we have created our own type.

Reply:
Yeah thats sensible. Yeah sorry thats what I meant, that its an existing JS array method name, its not a problem as you said, but your change makes it more readable.